pkg/util/priority/priority_test.go

```diff
@@ -157,6 +162,203 @@ func TestGetPriorityFromPriorityClass(t *testing.T) {
 				t.Errorf("unexpected name: got: %s, expected: %s", name, tt.wantPriorityClassName)
 			}
 
+			if source != tt.wantSource {
+				t.Errorf("unexpected source: got: %s, expected: %s", source, tt.wantSource)
+			}
+
+			if value != tt.wantPriorityClassValue {
+				t.Errorf("unexpected value: got: %d, expected: %d", value, tt.wantPriorityClassValue)
+			}
+		})
+	}
+}
+
+func TestGetPriorityFromWorkloadPriorityClass(t *testing.T) {
+	scheme := runtime.NewScheme()
+	if err := kueue.AddToScheme(scheme); err != nil {
+		t.Fatalf("Failed adding kueue scheme: %v", err)
+	}
+
+	tests := map[string]struct {
+		workloadPriorityClassList      *kueue.WorkloadPriorityClassList
+		workloadPriorityClassName      string
+		wantWorkloadPriorityClassName  string
+		wantWorkloadPriorityClassValue int32
+		wantSource                     string
+		wantErr                        string
+	}{
+		"workloadPriorityClass is specified and it exists": {
+			workloadPriorityClassList: &kueue.WorkloadPriorityClassList{
+				Items: []kueue.WorkloadPriorityClass{
+					{
+						ObjectMeta: v1.ObjectMeta{Name: "test"},
+						Value:      50,
+					},
+				},
+			},
+			workloadPriorityClassName:      "test",
+			wantWorkloadPriorityClassName:  "test",
+			wantSource:                     constants.WorkloadPriorityClassSource,
+			wantWorkloadPriorityClassValue: 50,
+		},
+		"workloadPriorityClass is specified and it does not exist": {
+			workloadPriorityClassList: &kueue.WorkloadPriorityClassList{
+				Items: []kueue.WorkloadPriorityClass{},
+			},
+			workloadPriorityClassName: "test",
+			wantSource:                "",
+			wantErr:                   `workloadpriorityclasses.kueue.x-k8s.io "test" not found`,
+		},
+	}
+
+	for desc, tt := range tests {
+		tt := tt
+		t.Run(desc, func(t *testing.T) {
+			t.Parallel()
+
+			builder := fake.NewClientBuilder().WithScheme(scheme).WithLists(tt.workloadPriorityClassList)
+			client := builder.Build()
+
+			name, value, source, err := GetPriorityFromWorkloadPriorityClass(context.Background(), client, tt.workloadPriorityClassName)
+			if tt.wantErr != "" {
+				if err == nil {
+					t.Fatalf("expected an error")
+				}
+
+				if diff := cmp.Diff(tt.wantErr, err.Error()); diff != "" {
+					t.Errorf("unexpected error (-want,+got):\n%s", diff)
+				}
+				return
+			}
```

Comment:
try to get this working with a single `cmp.Diff(tt.wantErr, err)`

you can use:

```golang
wantErr: apierrors.NewNotFound(..., ...)
```

https://github.com/kubernetes/apimachinery/blob/0d057e543013c79e20abb000df19bc16d286b777/pkg/api/errors/errors.go#L144